### PR TITLE
fix(blocker): fix flaky tests

### PR DESCRIPTION
**Pull request by vladopajic**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Closes:  #3416

### CODE REVIEW

File: pkg/blocker/blocker_test.go [hunk 1]
```diff
@@ -18,144 +17,116 @@ import (
...
+	mock := mockBlockLister(func(a swarm.Address, d time.Duration, r string) error {
+		t.Fatalf("address should not be blocked")
+
+		return nil
+	})
+
+	b := blocker.New(mock, flagTime, blockTime, time.Millisecond, nil, logger)
 	defer b.Close()
 
+	// Flagging address shouldn't block it imidietly
```
Comment (notanatol) [inline pkg/blocker/blocker_test.go:119]: comment typo `immediately`

**Comment by notanatol** [comment]
is there a way to confirm that these fixes remove flakiness? Or do we just run them with a large `-count` option?

**Comment by vladopajic** [comment]
@notanatol 

All fixes which involve `time.Sleep` as tool for waiting for some async background task (aka some goroutine), are almost never real solution. All they do is increase chance for this task to happen sooner then goroutine being blocked - but this is never guaranteed.

Imagine having two goroutines A (currently blocked by `time.Sleep`) and B (doing some task and result of this task is expected by A). In normal circumstances B might finish way sooner then A attempts to read B's result (A block interval > B work interval). Problem arises when B's goroutine gets suspended (for whichever reason); in which case B's work interval gets increased (sometimes undesirable more then A's block interval).

Afaik there is not way to tests this as all goroutines scheduling is done by go runtime and no mechanism is exposed to the user. This wouldn't be surprise because it was never expected of user to write synchronization in such way. Instead, B should communicated back (via channels) result to the A.

Only thing we can do is increase likelihood for suspending any goroutine (by making tests parallel with high count)

**Review by notanatol — APPROVED** [review]

**Comment by notanatol** [comment]
> @notanatol
> 
> All fixes which involve `time.Sleep` as tool for waiting for some async background task (aka some goroutine), are almost never real solution. All they do is increase chance for this task to happen sooner then goroutine being blocked - but this is never guaranteed.
> 
> Imagine having two goroutines A (currently blocked by `time.Sleep`) and B (doing some task and result of this task is expected by A). In normal circumstances B might finish way sooner then A attempts to read B's result (A block interval > B work interval). Problem arises when B's goroutine gets suspended (for whichever reason); in which case B's work interval gets increased (sometimes undesirable more then A's block interval).
> 
> Afaik there is not way to tests this as all goroutines scheduling is done by go runtime and no mechanism is exposed to the user. This wouldn't be surprise because it was never expected of user to write synchronization in such way. Instead, B should communicated back (via channels) result to the A.
> 
> Only thing we can do is increase likelihood for suspending any goroutine (by making tests parallel with high count)

yeah that's fair enough

**Pull request merged**